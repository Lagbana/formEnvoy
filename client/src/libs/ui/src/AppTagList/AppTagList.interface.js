export interface AppTagType<T> {
  label: string;
  value: T;
}
