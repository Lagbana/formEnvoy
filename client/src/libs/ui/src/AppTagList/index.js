export * from './AppTagList';
export * from './AppTagList.interface';
