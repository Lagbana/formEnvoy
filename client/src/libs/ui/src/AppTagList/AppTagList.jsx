import React from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';
import { AppTag } from '../AppTagList/components/AppTag';
import { AppTagType } from './AppTagList.interface';

export interface AppTagListProps<T> {
  tags: AppTagType<T>[];
  onRemove?: (tag: AppTagType<T>) => void;
  editable?: boolean;
  style?: ViewStyle;
}

export function AppTagList<T>({
  tags = [],
  onRemove,
  editable = true,
  style,
}: AppTagListProps<T>) {

  const styles = StyleSheet.create({
    container: {
      display: 'flex',
      flexDirection: 'row',
      flexWrap: 'wrap',
      marginVertical: 20,
    },
	});

  return (
    <View style={[styles.container, style]}>
      {tags.map((item, index) => (
        <AppTag
          key={index}
          tag={item}
          onPress={() => { if (onRemove) { onRemove(item) } }}
          editable={editable}
        />
      ))}
    </View>
  );
};
