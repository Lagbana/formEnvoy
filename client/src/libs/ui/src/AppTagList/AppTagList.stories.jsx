import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppTagList } from './AppTagList'

const tags = [
  { label: 'first tag', value: 100 },
  { label: 'second tag', value: 200 },
  { label: 'third tag', value: 300 }
]

storiesOf('AppTagList', module)
  .add('Editable: True', () => <AppTagList editable={true} tags={tags} />)
  .add('Editable: False', () => <AppTagList editable={false} tags={tags} />)
