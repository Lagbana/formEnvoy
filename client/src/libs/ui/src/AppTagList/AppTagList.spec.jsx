import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppTagList } from './AppTagList'

configure({ adapter: new Adapter() })

test('<AppTagList /> renders correctly', () => {
  const component = shallow(
    <AppTagList tags={[{ label: 'test label', value: 0 }]} />
  )
  expect(toJson(component)).toMatchSnapshot()
})
