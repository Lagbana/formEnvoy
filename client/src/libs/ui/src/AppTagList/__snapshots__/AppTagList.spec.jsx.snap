// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppTagList /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginVertical": 20,
      },
      undefined,
    ]
  }
>
  <AppTag
    editable={true}
    key="0"
    onPress={[Function]}
    tag={
      Object {
        "label": "test label",
        "value": 0,
      }
    }
  />
</View>
`;
